.dotnet/src/Custom/Internal/ClientPipelineExtensions.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Runtime.ExceptionServices;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OpenAI;
+
+internal static partial class ClientPipelineExtensions
+{
+    // CUSTOM:
+    // - Supplemented exception body with deserialized OpenAI error details
+
+    public static async ValueTask<PipelineResponse> ProcessMessageAsync(
+        this ClientPipeline pipeline,
+        PipelineMessage message,
+        RequestOptions options)
+    {
+        await pipeline.SendAsync(message).ConfigureAwait(false);
+
+        if (message.Response.IsError && (options?.ErrorOptions & ClientErrorBehaviors.NoThrow) != ClientErrorBehaviors.NoThrow)
+        {
+            throw await TryBufferResponseAndCreateErrorAsync(message).ConfigureAwait(false) switch
+            {
+                string errorMessage when !string.IsNullOrEmpty(errorMessage)
+                    => new ClientResultException(errorMessage, message.Response),
+                _ => new ClientResultException(message.Response),
+            };
+        }
+
+        return message.Response;
+    }
+        
+    public static PipelineResponse ProcessMessage(
+        this ClientPipeline pipeline,
+        PipelineMessage message,
+        RequestOptions options)
+    {
+        pipeline.Send(message);
+
+        if (message.Response.IsError && (options?.ErrorOptions & ClientErrorBehaviors.NoThrow) != ClientErrorBehaviors.NoThrow)
+        {
+            throw TryBufferResponseAndCreateError(message) switch
+            {
+                string errorMessage when !string.IsNullOrEmpty(errorMessage)
+                    => new ClientResultException(errorMessage, message.Response),
+                _ => new ClientResultException(message.Response),
+            };
+        }
+
+        return message.Response;
```

Comment:
This will need to add the lines from https://github.com/Azure/autorest.csharp/issues/4693 if they aren't able to pick them up from the generator.